File: baekjoon/stage04/1110_더하기_사이클.py
import sys

n = int(sys.stdin.readline())
cycle = 0

new_n = n
while True:
    new_n = (new_n%10)*10 + (int(new_n/10) + new_n%10)%10
    cycle = cycle + 1
    if new_n == n:
        break
print(cycle)

File: problem-solving/baekjoon/step04/1110_더하기_사이클.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	var n, cycle int
	reader := bufio.NewReader(os.Stdin)
	fmt.Fscanln(reader, &n)

	var newN = n
	for true {
		cycle++
		var sum int
		if newN/10 == 0 {
			sum = newN
		} else {
			sum = newN/10 + newN%10
		}
		newN = newN%10*10 + sum%10
		if newN == n {
			break
		}
	}
	fmt.Println(cycle)
}
